src/idl_gen_swift.cpp

```diff
@@ -0,0 +1,721 @@
+      std::string offset = "{{ACCESS}}.postion + {{OFFSET}}") {
+    return "{{VALUETYPE}}({{ACCESS}}.bb, o: " + offset + ") }";
+  }
+
+  std::string GenMutate(std::string offset, std::string get_offset = "",
+                        bool isRaw = false) {
+    return "\tpublic func mutate({{VALUENAME}}: {{VALUETYPE}}) -> Bool {" +
+           get_offset + " return {{ACCESS}}.mutate({{VALUENAME}}" +
+           (isRaw ? ".rawValue" : "") + ", index: " + offset + ") }";
+  }
+
+  std::string GenMutateArray() {
+    return "\tpublic func mutate({{VALUENAME}}: {{VALUETYPE}}, at index: "
+           "Int32) -> Bool { " +
+           GenOffset() +
+           "return {{ACCESS}}.directMutate({{VALUENAME}}, index: "
+           "{{ACCESS}}.vector(at: o) + index * {{SIZE}}) }";
+  }
+
+  std::string GenEnumBaseContructor() {
+    return "{{VALUETYPE}}(rawValue: {{CONSTANT}})!";
+  }
+
+  std::string GenEnumConstructor(std::string at = "{{OFFSET}}") {
+    return "{{VALUETYPE}}(rawValue: " + GenReader("BASEVALUE", at) + ") ";
+  }
+
+  std::string GenType(const Type &type) const {
+    return IsScalar(type.base_type)
+               ? GenTypeBasic(type)
+               : (IsArray(type) ? GenType(type.VectorType())
+                                : GenTypePointer(type));
+  }
+
+  std::string GenTypePointer(const Type &type) const {
+    switch (type.base_type) {
+      case BASE_TYPE_STRING: return "String";
+      case BASE_TYPE_VECTOR: return GenType(type.VectorType());
+      case BASE_TYPE_STRUCT: return WrapInNameSpace(*type.struct_def);
+      case BASE_TYPE_UNION:
+      default: return "FlatBufferObject";
+    }
+  }
+
+  std::string GenTypeBasic(const Type &type) const {
+    return GenTypeBasic(type, true);
+  }
+
+  std::string GenTypeBasic(const Type &type, bool can_override) const {
+    // clang-format off
+    static const char * const swift_type[] = {
+      #define FLATBUFFERS_TD(ENUM, IDLTYPE, \
+              CTYPE, JTYPE, GTYPE, NTYPE, PTYPE, RTYPE, KTYPE, STYPE) \
+        #STYPE,
+        FLATBUFFERS_GEN_TYPES(FLATBUFFERS_TD)
+      #undef FLATBUFFERS_TD
+    };
+    // clang-format on
+    if (can_override) {
+      if (type.enum_def)
+        return WrapInNameSpace(type.enum_def->defined_namespace,
+                               Name(*type.enum_def));
+      if (type.base_type == BASE_TYPE_BOOL) return "Bool";
+    }
+    return swift_type[type.base_type];
+  }
+
+  std::string GenEnumDecl(const EnumDef &enum_def) const {
+    return "enum " + Name(enum_def);
+  }
+
+  std::string EscapeKeyword(const std::string &name) const {
+    return keywords_.find(name) == keywords_.end() ? name : name + "_";
+  }
+
+  std::string Name(const EnumVal &ev) const { return EscapeKeyword(ev.name); }
+
+  std::string Name(const Definition &def) const {
+    return EscapeKeyword(MakeCamel(def.name, false));
+  }
+
+  void LowerCase(std::string *value) {
+    transform(value->begin(), value->end(), value->begin(), ::tolower);
```

Comment:
MSVC2019 compilation failed with error:
>include\algorithm(1310,24): error C2220: the following warning is treated as an error
>warning C4244: '=': conversion from 'int' to 'char', possible loss of data

You can see ToUpper implementation in idl_gen_cpp or [here](https://en.cppreference.com/w/cpp/string/byte/tolower).
>Like all other functions from <cctype>, the behavior of std::tolower is undefined if the argument's value is neither representable as unsigned char nor equal to EOF. To use these functions safely with plain chars (or signed chars), the argument should first be converted to unsigned char